pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,353 @@
+    in python 2
+        @abstractproperty
+    """
+
+    if six.PY3:
+        return property(abstractmethod(func))
+    else:
+        return abstractproperty(func)
+
+
+@six.add_metaclass(ABCMeta)
+class ObjectStoreAPI(object):
+    """
+    This abstract class defines the APIs for the methods of an object store
+    for CIM objects including CIMClass, CIMInstance, and CIMQualifierDeclaration
+    objectsthat constitute a WBEM server repository.  This
+    class provides the abstract methods for creating, accessing, and deleting,
+    CIM objects of a single CIM object type in the repository.
+
+    These APIs allows creating, updating, deleting, and retrieving these
+    objects.
+    """
+
+    def __init__(self, case_insensitive_names, cim_object_type):
+        """
+        Initialize the object store.
+
+        Parameters:
+
+            case_insensitive-names(:class:`py:bool`):
+              If True, the names uniquely identify objects in the object store
+              case insensitively. The names must be strings.
+              If False, names identify objects. The names may be strings
+              or other objects that can be used to uniquely identify
+              objects in the object store (ex. CIMInstanceName).
+
+            cim_object_type(:term:`string`):
+              The Pywbem cim object type as defined in cim_types.py for the
+              objects in the data store. Used to verify values on create.
+        """
+        self._case_insensitive_names = case_insensitive_names
+        self._cim_object_type = cim_object_type
+
+    @abstractmethod
+    def exists(self, name):
+        """
+        Test if cim_object defined by name exists in the
+        object store
+
+        Paramsters
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+        Returns:
+            True if the object exists; False if it does not exist
+        """
+        pass
+
+    @abstractmethod
+    def get(self, name, copy=True):
+        """
+        Return the cim_object defined by name and namespace if it exists.
+
+        Parameters:
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+          copy: If True, insures that modifying the returned object does not
+            change the data store. The default behavior is True.  If
+            copy is False, the object in the object store is returned but
+            if it is modified by the user, the object in the store may
+            be modified also.
+
+        Returns:
+            Returns the cim_object identified by the name parameter
+
+        Raises:
+            KeyError: Object not in this object store
+        """
+        pass
+
+    @abstractmethod
+    def create(self, name, cim_object):
+        """
+        Adds the cim_object identified by name to the object store.
```

Comment:
It is not clear from the description whether a copy is made. I.e. if the caller modified the object provided to create(), does it modify the object in the object store?

Reply:
Yes it always copies. Added comment. DONE